java-agent/instrumentation/reactor-core-3.1/src/main/java8/io/opentelemetry/auto/instrumentation/reactor/core/ReactorCoreAdviceUtils.java

```diff
@@ -39,16 +42,16 @@
   }
 
   public static void finishSpanIfPresent(final Context context, final Throwable throwable) {
-    finishSpanIfPresent(context.getOrDefault(PUBLISHER_CONTEXT_KEY, (AgentSpan) null), throwable);
+    finishSpanIfPresent(context.getOrDefault(PUBLISHER_CONTEXT_KEY, (Span) null), throwable);
   }
 
-  public static void finishSpanIfPresent(final AgentSpan span, final Throwable throwable) {
+  public static void finishSpanIfPresent(final Span span, final Throwable throwable) {
     if (span != null) {
       if (throwable != null) {
-        span.setError(true);
-        span.addThrowable(throwable);
+        span.setStatus(Status.UNKNOWN);
+        DECORATE.addThrowable(span, throwable);
```

Comment:
```suggestion
        BaseDecorator.addThrowable(span, throwable);
```
and then decorator is not needed